File: backend/app/schemas/agent.py
"""后端规则 Agent 的 HTTP 请求与响应模型。"""

from __future__ import annotations

from typing import Any, Literal

from pydantic import BaseModel, Field


class AgentChatRequest(BaseModel):
    text: str = Field(min_length=1)


class AgentConfirmRequest(BaseModel):
    preview_id: str = Field(min_length=1)


class AgentModelUsage(BaseModel):
    used: bool = False
    provider: str | None = None
    name: str | None = None
    fallback_reason: str | None = None


class PreviewEnvelope(BaseModel):
    preview_id: str
    type: Literal["application", "interview", "status_update"]
    fields: dict[str, Any]
    missing: dict[str, list[str]] | None = None
    warnings: list[str] = Field(default_factory=list)


class AgentResponse(BaseModel):
    ok: bool
    intent: str
    message: str
    data: dict[str, Any] | None = None
    preview: PreviewEnvelope | None = None
    requires_confirmation: bool = False
    model: AgentModelUsage = Field(default_factory=AgentModelUsage)

File: frontend/src/api/agent.ts
import { apiPost } from "./client";

export type AgentPreviewType = "application" | "interview" | "status_update";

export interface AgentModelUsage {
  used: boolean;
  provider: string | null;
  name: string | null;
  fallback_reason?: string | null;
}

export interface AgentMissingFields {
  required: string[];
  recommended: string[];
}

export interface AgentPreview {
  preview_id: string;
  type: AgentPreviewType;
  fields: Record<string, unknown>;
  missing: AgentMissingFields | null;
  warnings: string[];
}

export interface AgentResponse {
  ok: boolean;
  intent: string;
  message: string;
  data: Record<string, unknown> | null;
  preview: AgentPreview | null;
  requires_confirmation: boolean;
  model: AgentModelUsage;
}

export function sendAgentMessage(text: string) {
  return apiPost<AgentResponse, { text: string }>("/agent/chat", { text });
}

export function confirmAgentPreview(previewId: string) {
  return apiPost<AgentResponse, { preview_id: string }>("/agent/confirm", {
    preview_id: previewId,
  });
}
